src/gui/symbology-ng/qgssymbollayerv2widget.cpp

```diff
@@ -110,56 +119,70 @@ void QgsSimpleLineSymbolLayerV2Widget::setSymbolLayer( QgsSymbolLayerV2* layer )
-  cboJoinStyle->setPenJoinStyle( mLayer->penJoinStyle() );
-  cboCapStyle->setPenCapStyle( mLayer->penCapStyle() );
-  cboPenStyle->blockSignals( false );
-  cboJoinStyle->blockSignals( false );
-  cboCapStyle->blockSignals( false );
+  QGS_BLOCK_SIGNAL_DURING_CALL( spinWidth, setValue( mLayer->width() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( btnChangeColor, setColor( mLayer->color() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( spinOffset, setValue( mLayer->offset() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboPenStyle, setPenStyle( mLayer->penStyle() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboJoinStyle, setPenJoinStyle( mLayer->penJoinStyle() ) )
+  QGS_BLOCK_SIGNAL_DURING_CALL( cboCapStyle, setPenCapStyle( mLayer->penCapStyle() ) )
 
   //use a custom dash pattern?
-  bool useCustomDashPattern = mLayer->useCustomDashPattern();
+  const bool useCustomDashPattern = mLayer->useCustomDashPattern();
   mChangePatternButton->setEnabled( useCustomDashPattern );
+  mDashPatternDDBtn->setEnabled( useCustomDashPattern );
   label_3->setEnabled( !useCustomDashPattern );
   cboPenStyle->setEnabled( !useCustomDashPattern );
-  mCustomCheckBox->blockSignals( true );
-  mCustomCheckBox->setCheckState( useCustomDashPattern ? Qt::Checked : Qt::Unchecked );
-  mCustomCheckBox->blockSignals( false );
+  QGS_BLOCK_SIGNAL_DURING_CALL( mCustomCheckBox, setCheckState( useCustomDashPattern ? Qt::Checked : Qt::Unchecked ) )
 
   //draw inside polygon?
-  bool drawInsidePolygon = mLayer->drawInsidePolygon();
-  mDrawInsideCheckBox->blockSignals( true );
-  mDrawInsideCheckBox->setCheckState( drawInsidePolygon ? Qt::Checked : Qt::Unchecked );
-  mDrawInsideCheckBox->blockSignals( false );
+  const bool drawInsidePolygon = mLayer->drawInsidePolygon();
+  QGS_BLOCK_SIGNAL_DURING_CALL( mDrawInsideCheckBox, setCheckState( drawInsidePolygon ? Qt::Checked : Qt::Unchecked ) )
 
   updatePatternIcon();
+
+  mDataDefinedPropertyButtons.clear();
+
+#define QGS_REGISTER_DATA_DEFINED_BUTTON( button, name, type, description )\
+  {\
+    QgsDataDefined dd( mLayer->dataDefinedProperty( name ) );\
+    button->init( mVectorLayer, &dd, QgsDataDefinedButton::type, Qt::escape( description ) );\
+    mDataDefinedPropertyButtons.insert( name, button );\
+  }
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mColorDDBtn, "color", String, QgsDataDefinedSymbolDialog::colorHelpText())
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mPenWidthDDBtn, "width", Double, "")
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mOffsetDDBtn, "offset", String, QgsDataDefinedSymbolDialog::offsetHelpText())
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mDashPatternDDBtn, "customdash", String, tr( "'<dash>;<space>' e.g. '8;2;1;2'" ) )
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mJoinStyleDDBtn, "joinstyle", String, tr( "'bevel'|'miter'|'round'" ))
+  QGS_REGISTER_DATA_DEFINED_BUTTON( mCapStyleDDBtn, "capstyle", String, tr( "'square'|'flat'|'round'" ) )
+#undef QGS_REGISTER_DATA_DEFINED_BUTTON
+
+  for (QMap< QString, QgsDataDefinedButton* >::const_iterator it 
```

Comment:
@vmora this code segment should also be implemented as a protected function in QgsSymbolLayerV2Widget to avoid duplication